strapi/strapi
# fix(data-transfer): sequential payload consumption

Hello,

Here's a fix PR for data transfers.

Thank you.

### What does it do?

It fixes an issue with data transfer when two stream payloads are received at the same time, and one of them contains the "end" action for an asset transfer, while the other still has "stream" actions.
There is a 100ms sleep looping while waiting for the previous payload to finish being read, this delay can be changed if you prefer another value.

### Why is it needed?

It solves data transfer in some cases with bigger files.

### How to test it?

The repo I tested this on is private and owned by my client. But I guess you could create a remote strapi instance with big media (videos > 50MB) and check if that happens for you.
Without this fix, it's a dice roll, it sometimes works and sometimes not, but with this, it's working in all cases.

### Related issue(s)/PR(s)

I'm guessing a lot of data transfer issues could be related to this fix, but I didn't check.
